### PR TITLE
[Merge on Monday 6.21.] Use new nav schema to render chrome nav

### PR DESCRIPTION
**Pull request by Hyperkid123**
jira: https://issues.redhat.com/browse/RHCLOUD-14154, https://issues.redhat.com/browse/RHCLOUD-14840

merge only after: https://github.com/RedHatInsights/cloud-services-config/pull/666 and https://github.com/RedHatInsights/cloud-services-config/pull/680

I have limited the release only to CI beta so we don't break every QA test if something goes wrong.

### TO DO
- [x] landing page nav
- [x] main nav
  - [x] rhel nav
  - [x] ansbile nav
  - [x] openshift nav
  - [x] managed services nav
  - [x] settings nav
  - [x] user preferences nav
- [x] enable client side routing
  - [x] add react routes for chrome 2 apps
  - [x] add common route for non modules
  - [x] use router link for nav links
  - [x] enable navigation between chrome 2 apps
  - [x] enable navigation in legacy apps 
  - [x] full page refresh between bundles
- [x] enable permissions
- [x] enable nav items highlights
- [x] deploy the new schemas (only in ci for now)
- [x] add extra links to schemas
- [x] properly location metadata
  - [x] active location
  - [x] active app
  - [x] active bundle
  - [x] page action
  - [x] page object id
- [x] clean up
  - [x] tests
  - [x] linter
  - [x] remove dead code
  
cc @john-dupuy

### CODE REVIEW

File: src/js/redux/reducers.js [hunk 1]
```diff
@@ -123,21 +123,76 @@ export function onPageObjectId(state, { payload }) {
...
+}
+
+export function loadModulesSchemaReducer(state, { payload: { schema } }) {
+  const scalprumConfig = Object.entries(schema).reduce(
+    (acc, [name, config]) => ({
+      ...acc,
+      [name]: {
+        name,
+        module: `${name}#./RootApp`,
+        manifestLocation: `${window.location.origin}${isBeta() ? '/beta' : ''}${config.manifestLocation}`,
```
Comment (karelhala) [inline src/js/redux/reducers.js:176]: I think we could have precalculated `manifestLocation` from `name` if it's missing from config.

**Review by karelhala — APPROVED** [review]

File: src/js/redux/reducers.js (hunk 1 above)
Comment (Hyperkid123) [reply src/js/redux/reducers.js:176]: I would like to avoid any calculation if possible. The module definition is not something that we will be creating/updating every few minutes. It will stay the same probably once an application is deployed for the first time. If the location is strictly set, we don't have to fear any potential edge cases in the future. No calculation equals no bugs.

**Review by rvsia — APPROVED** [review]
Didn't test it in the UI, but code looks solid and the tests are awesome! 🏆

File: src/js/App/Routes/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          .flat(),
+      ],
+      []
+    )
+    .sort((a, b) => (a.path.length < b.path.length ? 1 : -1));
+  return reactModules;
+};
+
+const Routes = ({ insightsContentRef }) => {
+  const modules = useSelector(({ chrome: { modules } }) => modules);
```
[the remark this answers is not in the document]
Comment (Hyperkid123) [reply src/js/App/Routes/index.js:33]: No we don't because anything under RootApp is not rendered until chrome was successfully loaded.

File: src/js/App/RootApp.js [hunk 3]
```diff
@@ -181,10 +108,10 @@ RootApp.propTypes = {
   config: PropTypes.any,
 };
 
-function stateToProps({ chrome: { activeApp, activeLocation, appId, pageAction, pageObjectId }, globalFilter: { globalFilterRemoved } = {} }) {
-  return { activeApp, activeLocation, appId, pageAction, pageObjectId, globalFilterRemoved };
+function stateToProps({ chrome: { activeLocation }, globalFilter: { globalFilterRemoved } = {} }) {
```
[the remark this answers is not in the document]
Comment (Hyperkid123) [reply src/js/App/RootApp.js:111]: I was thinking the same but I would like to keep it for a general refactoring round after this is done.

**Comment by rvsia** [comment]
The nested applications are not selected

**Before**

Screenshot 2021-06-15 at 16 52 05

**After**

Screenshot 2021-06-15 at 16 51 59

**Comment by rvsia** [comment]
The documentation link is missing

**Before**

Screenshot 2021-06-15 at 16 52 49

**After**

Screenshot 2021-06-15 at 16 52 53

**Comment by Hyperkid123** [comment]
> The nested applications are not selected

@rvsia everywhere or just in applications?

**Comment by rvsia** [comment]
https://ci.foo.redhat.com:1337/beta/ansible/ansible-dashboard fails

```jsx
TypeError: Cannot read property 'init' of undefined
    at Object.<anonymous> (471.f5f789fcab280be53c4b.js:2)
    at 471.f5f789fcab280be53c4b.js:2
    at Object.next (471.f5f789fcab280be53c4b.js:2)
    at a (471.f5f789fcab280be53c4b.js:2)
```

And the whole section does not work for me.


Screenshot 2021-06-15 at 16 58 35

**Comment by rvsia** [comment]
@Hyperkid123 seems that only in settings

**Comment by Hyperkid123** [comment]
> @Hyperkid123 seems that only in settings

I suspect that the notifications app is breaking it. See some huge memory spikes and it overrides the URL always to `/notifications`. I think they have some history listener which is not cleared on unmount. I will take a look into it.

**Comment by Hyperkid123** [comment]
> And the whole section does not work for me.

@rvsia I am unable to reproduce the left nav not rendering problem. I have found the issue with the dashboard though and working on a fix.

**Comment by Hyperkid123** [comment]
@rvsia the nav highlight issues in settings should be fixed by this PR: https://github.com/RedHatInsights/notifications-frontend/pull/130

**Comment by Hyperkid123** [comment]
@rvsia the automation hub links should be fixed by this PR: https://github.com/RedHatInsights/cloud-services-config/pull/680 ad this commit: https://github.com/RedHatInsights/insights-chrome/pull/1460/commits/9b609ac96b9b34cc7ab1e0a49566c7022d808be4

**Review by rvsia — APPROVED** [review]
Tested in the UI. All my previous findings were fixed! Thank you @Hyperkid123 ! 🏆
